ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml

```diff
@@ -347,6 +347,37 @@ let dss_loadavg () =
     )
   ]
 
+let dss_hostload xc domains =
+  let physinfo = Xenctrl.physinfo xc in
+  let pcpus = physinfo.Xenctrl.nr_cpus in
+  let rec loop n f = if n < 0 then 0 else f n + loop (n - 1) f in
+  let load =
+    List.fold_left
+      (fun acc (dom, _, domid) ->
```

Comment:
Not sure if it is possible to read stale data and hence get error or exception. 
I would suggest to ignore the potential errors and exceptions, instead of populate them.

Reply:
I checked other ds using `Xenctrl`, they do not catch exception either.

Reply:
I think they are catching. https://github.com/xapi-project/xen-api/blob/fa18bc0ace9932011058a1190b34d04c080e56bf/ocaml/xcp-rrdd/bin/rrdd/xcp_rrdd.ml#L292

And this could be confirmed in https://code.citrite.net/projects/XSU/repos/xen/browse/tools/ocaml/libs/xc/xenctrl_stubs.c#522 

So I think it would be better to catch the exceptions.

Reply:
`domain_get_vcpuinfo` may raise exception indeed.
I checked following code to call the ds generator which already handled all exception and return `[]` ds
```
let generate_all_dom0_stats xc timestamp domains =
  let handle_generator (name, generator) =
    (name, handle_exn name (fun _ -> generator xc timestamp domains) [])
  in
  List.map handle_generator dom0_stat_generators
```